### PR TITLE
Improved filtering on empty values.

### PR DESCRIPTION
**Pull request by areksolek**
Improved filtering on empty values.

### CODE REVIEW

**Comment by ghost** [comment]
## Black Duck Security Report

Merging #15265 into developer will not change security risk.

[Click here to see full report](https://copilot.blackducksoftware.com/github/repos/YetiForceCompany/YetiForceCRM/pull-requests/15265)

File: layouts/basic/modules/Vtiger/List/Field/Base.tpl [hunk 1]
```diff
@@ -36,10 +36,10 @@
 	{else}
 		<div class="input-group">
 			<input type="text" name="{$FIELD_MODEL->getName()}"  {if !empty($FIELD_MODEL->get('source_field_name'))} data-source-field-name="{$FIELD_MODEL->get('source_field_name')}" data-module-name="{$FIELD_MODEL->getModuleName()}"
-			{/if} class="listSearchContributor form-control" value="{$SEARCH_VALUE}" title='{$LABEL['label']}' data-fieldinfo='{$FIELD_INFO|escape}' {if !$FIELD_MODEL->searchLockedFields() || !$FIELD_MODEL->isActiveSearchView()}disabled{/if}/>
+			{/if} class="listSearchContributor form-control" value="{$SEARCH_VALUE}" title='{$LABEL['label']}' data-fieldinfo='{$FIELD_INFO|escape}' {if !$FIELD_MODEL->searchLockedEmptyFields() || !$FIELD_MODEL->isActiveSearchView()}disabled{/if}/>
 			<div class="input-group-append">
 				<div class="input-group-text px-1">
-					<input type="checkbox" class="js-empty-value" {if !$FIELD_MODEL->searchLockedFields() || !$FIELD_MODEL->isActiveSearchView()} checked {/if} {if !$FIELD_MODEL->isActiveSearchView()}disabled{/if} >
+					<input type="checkbox" class="js-empty-value" {if !$FIELD_MODEL->searchLockedEmptyFields()} checked {/if} {if !$FIELD_MODEL->isActiveSearchView()}disabled{/if} >
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Vtiger/List/Field/Base.tpl]:
```suggestion
					<input type="checkbox" class="js-empty-value" {if !$FIELD_MODEL->searchLockedEmptyFields()}checked{/if} {if !$FIELD_MODEL->isActiveSearchView()}disabled{/if} >
```

File: layouts/basic/modules/Vtiger/ListViewContents.tpl [hunk 2]
```diff
@@ -24,7 +24,7 @@
 	<input type="hidden" id="excludedIds" name="excludedIds"/>
 	<input type='hidden' value="{$PAGING_MODEL->getPageLimit()}" id='pageLimit'>
 	<input type="hidden" value="{$LISTVIEW_ENTRIES_COUNT}" id="noOfEntries">
-	<input type="hidden" class="js-locked-fields" data-js="value" value="{\App\Purifier::encodeHtml($LOCKED_FIELDS)}"/>
+	<input type="hidden" class="js-empty-fields" data-js="value" value="{\App\Purifier::encodeHtml($LOCKED_FIELDS)}"/>
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Vtiger/ListViewContents.tpl]:
```suggestion
	<input type="hidden" class="js-empty-fields" data-js="value" value="{\App\Purifier::encodeHtml(\App\Json::encode($LOCKED_FIELDS))}"/>
```

File: modules/Vtiger/views/List.php [hunk 3]
```diff
@@ -283,5 +283,6 @@ public function initializeListViewContents(App\Request $request, Vtiger_Viewer $
 		$viewer->assign('IS_MODULE_DELETABLE', $this->listViewModel->getModule()->isPermitted('Delete'));
 		$viewer->assign('SEARCH_DETAILS', $searchParams);
 		$viewer->assign('SEARCH_PARAMS', $searchParamsRaw);
+		$viewer->assign('LOCKED_EMPTY_FIELDS', $request->isEmpty('lockedEmptyFields', true) ? false : \App\Json::encode($request->getArray('lockedEmptyFields')));
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/views/List.php]:
```suggestion
		$viewer->assign('LOCKED_EMPTY_FIELDS', $request->isEmpty('lockedEmptyFields', true) ? []: $request->getArray('lockedEmptyFields'));
```

File: modules/Vtiger/views/RecordsList.php [hunk 4]
```diff
@@ -261,6 +260,7 @@ public function initializeContent(App\Request $request)
 		$viewer->assign('RECORD_SELECTED', $request->getBoolean('record_selected', false));
 		$viewer->assign('CUSTOM_VIEWS', CustomView_Record_Model::getAllByGroup($request->getModule()));
 		$viewer->assign('LOCKED_FIELDS', $request->isEmpty('lockedFields', true) ? false : \App\Json::encode($request->getArray('lockedFields')));
+		$viewer->assign('LOCKED_EMPTY_FIELDS', $request->isEmpty('lockedEmptyFields', true) ? false : \App\Json::encode($request->getArray('lockedEmptyFields')));
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/views/RecordsList.php]:
```suggestion
		$viewer->assign('LOCKED_EMPTY_FIELDS', $request->isEmpty('lockedEmptyFields', true) ? [] : $request->getArray('lockedEmptyFields'));
```

File: modules/Vtiger/views/RelatedList.php [hunk 5]
```diff
@@ -175,7 +175,7 @@ public function process(App\Request $request)
 		$viewer->assign('SEARCH_PARAMS', $searchParamsRaw);
 		$viewer->assign('VIEW', $request->getByType('view'));
 		$viewer->assign('SHOW_RELATED_WIDGETS', \in_array($relationModel->getId(), App\Config::module($moduleName, 'showRelatedWidgetsByDefault', [])));
-		$viewer->assign('LOCKED_FIELDS', $request->isEmpty('lockedFields', true) ? false : \App\Json::encode($request->getArray('lockedFields')));
+		$viewer->assign('LOCKED_EMPTY_FIELDS', $request->isEmpty('lockedEmptyFields', true) ? false : \App\Json::encode($request->getArray('lockedEmptyFields')));
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/views/RelatedList.php]: ?

File: layouts/basic/modules/Vtiger/RelatedList.tpl [hunk 6]
```diff
@@ -29,7 +29,7 @@
 	<input type="hidden" id="tab_label" value="{\App\Purifier::encodeHtml($RELATION_MODEL->get('label'))}"/>
 	<input type="hidden" id="relationId" value="{$RELATION_MODEL->getId()}"/>
 	<input type="hidden" id="search_params" value="{\App\Purifier::encodeHtml(\App\Json::encode($SEARCH_PARAMS))}">
-	<input type="hidden" class="js-locked-fields" data-js="value" value="{\App\Purifier::encodeHtml($LOCKED_FIELDS)}"/>
+	<input type="hidden" class="js-empty-fields" data-js="value" value="{\App\Purifier::encodeHtml($LOCKED_EMPTY_FIELDS)}"/>
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Vtiger/RelatedList.tpl]: ?

File: layouts/basic/modules/Vtiger/Modals/RecordsList.tpl [hunk 7]
```diff
@@ -18,6 +18,7 @@
 		<input type="hidden" id="search_params" value="{\App\Purifier::encodeHtml(\App\Json::encode($SEARCH_PARAMS))}">
 		{include file=\App\Layout::getTemplatePath('ListViewAlphabet.tpl', $MODULE)}
 		<input type="hidden" class="js-locked-fields" data-js="value" value="{\App\Purifier::encodeHtml($LOCKED_FIELDS)}"/>
+		<input type="hidden" class="js-empty-fields" data-js="value" value="{\App\Purifier::encodeHtml($LOCKED_EMPTY_FIELDS)}"/>
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Vtiger/Modals/RecordsList.tpl]: ?

**Review by mariuszkrzaczkowski — CHANGES_REQUESTED** [review]

**Pull request merged**